### ISSUE TITLE
Linter should throw if passed a string configuration

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#1167]
#### Repro

I did a silly and forgot to call `JSON.parse` on the value being passed to `Linter`.

```typescript
// wrong
let linter = new Linter("file.ts", "...", "{ \"rules\": ... } ");
```

```typescript
// correct
let linter = new Linter('file.ts", "...", { rules: ... });
```

#### Actual behavior

It ran without error, but reported 0 lint failures.

#### Expected behavior

It should have blown up and told me I'm an idiot for passing an invalid tslint.json equivalent.

:(

*(I'll send a PR for this unless directed otherwise)*

**Comment by JKillian** [comment]
Seems like a good change to make to me 👍

**Comment by JoshuaKGoldberg** [comment]
:fireworks:
